src/native/libs/System.Security.Cryptography.Native/pal_evp_pkey_ml_dsa.c

```diff
@@ -0,0 +1,297 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_evp_pkey.h"
+#include "pal_evp_pkey_ml_dsa.h"
+#include "pal_utilities.h"
+#include "openssl.h"
+#include <assert.h>
+
+// ML-DSA-44 signature length is 2420 bytes.
+#define SHORTEST_POSSIBLE_ML_DSA_SIG_LEN 2420
+
+int32_t CryptoNative_IsSignatureAlgorithmAvailable(const char* algorithm)
+{
+#ifdef FEATURE_DISTRO_AGNOSTIC_SSL
+    if (!API_EXISTS(EVP_PKEY_sign_message_init) ||
+        !API_EXISTS(EVP_PKEY_verify_message_init))
+    {
+        return 0;
+    }
+#endif
+
+    int32_t ret = 0;
+#ifdef HAVE_OPENSSL_EVP_PKEY_SIGN_MESSAGE_INIT
+    EVP_SIGNATURE* sigAlg = NULL;
+
+    sigAlg = EVP_SIGNATURE_fetch(NULL, algorithm, NULL);
+    if (sigAlg)
+    {
+        ret = 1;
+        EVP_SIGNATURE_free(sigAlg);
+    } 
+#endif
+
+    (void)algorithm;
+    return ret;
+}
+
+EVP_PKEY* CryptoNative_MLDsaGenerateKey(const char* keyType, uint8_t* seed, int32_t seedLen)
+{
+#ifdef FEATURE_DISTRO_AGNOSTIC_SSL
+    if (!API_EXISTS(EVP_PKEY_sign_message_init) ||
+        !API_EXISTS(EVP_PKEY_verify_message_init))
+    {
+        return NULL;
+    }
+#endif
+
+#ifdef HAVE_OPENSSL_EVP_PKEY_SIGN_MESSAGE_INIT
+    ERR_clear_error();
+
+    if (seed && seedLen != 32)
+    {
+        return NULL;
+    }
+
+    EVP_PKEY_CTX* pctx = EVP_PKEY_CTX_new_from_name(NULL, keyType, NULL);
+    EVP_PKEY* pkey = NULL;
+
+    if (!pctx)
+    {
+        return NULL;
+    }
+
+    if (EVP_PKEY_keygen_init(pctx) <= 0)
+    {
+        goto done;
+    }
+
+    if (seed)
+    {
+        OSSL_PARAM params[2];
+        params[0] = OSSL_PARAM_construct_octet_string(OSSL_PKEY_PARAM_ML_DSA_SEED, (void*)seed, Int32ToSizeT(seedLen));
+        params[1] = OSSL_PARAM_construct_end();
```

Comment:
```suggestion
        OSSL_PARAM params[] =
        {
            OSSL_PARAM_construct_octet_string(OSSL_PKEY_PARAM_ML_DSA_SEED, (void*)seed, Int32ToSizeT(seedLen)),
            OSSL_PARAM_construct_end(),
        };
```

Let the compiler count for you :smile: